harness/determined/cli/dev.py

```diff
@@ -255,6 +258,33 @@ def list_bindings(args: Namespace) -> None:
         print(bindings_sig_str(name, params))
 
 
+def auto_complete_binding(available_calls: List[str], fn_name: str) -> str:
+    """
+    utility to allow partial matching of binding names.
+    """
+    if fn_name in available_calls:
+        return fn_name
+    matches = [n for n in available_calls if re.match(f".*{fn_name}.*", n, re.IGNORECASE)]
+    if not matches:
+        raise errors.CliError(f"no such binding found: {fn_name}")
+    if not sys.stdout.isatty():
+        raise errors.CliError(
+            f"no exact matches for '{fn_name}'. Did you mean:" + "\n{}".format("\n".join(matches))
+        )
+    indexed_matches = [f"{idx}: {match}" for idx, match in list(enumerate(matches))]
+    user_idx = (
```

Comment:
nit: selected_idx? user had me confused for the "user" concept in our cli/elsewhere